site/app/controllers/admin/ReportController.php

```diff
@@ -652,95 +684,209 @@ public function generateCustomization() {
                 JsonResponse::getErrorResponse($msg),
                 null,
                 new RedirectResponse($redirect_url)
             );
         }
 
-        $msg = 'Rainbow Grades Customization uploaded';
-        $this->core->addSuccessMessage($msg);
-        return new MultiResponse(
-            JsonResponse::getSuccessResponse([
-                'customization_path' => $rainbow_grades_dir
-            ]),
-            null,
-            new RedirectResponse($redirect_url)
-        );
+        // Set headers and read the file
+        header('Content-Type: application/json');
+        header("Content-Transfer-Encoding: Binary");
+        header("Content-disposition: attachment; filename=\"" . basename($file_path) . "\"");
+
+        readfile($file_path);
+        exit;
+        // Do not add return type, for example if you add :MultiResponse
+        // php-stan will start yelling that missing return statement
+        // if we add return statement, it will be written into the downloaded file as json
+        // then it will break the Make remove_json_comments, so no linting is OK here.
     }
 
-    #[Route("/courses/{_semester}/{_course}/reports/rainbow_grades_status")]
+
+    #[Route('/courses/{_semester}/{_course}/reports/rainbow_grades_customization/manual_or_gui', methods: ['POST'])]
+    public function setRainbowGradeCustomization(): JsonResponse {
+
+        // Extract the value from $_POST
+        $selectedValue = $_POST['selected_value'] ?? null;
+
+        if (!isset($selectedValue) || trim($selectedValue) === '') {
+            $msg = 'Invalid request: No selected value provided.';
+            return JsonResponse::getErrorResponse($msg);
+        }
+
+        $rainbow_grades_dir = FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), "rainbow_grades");
+        $customization_src = null;
+        $customization_dest = FileUtils::joinPaths($rainbow_grades_dir, 'customization.json');
+
+        // Determine the source file based on the selected value
+        switch ($selectedValue) {
+            case 'manual':
+                $customization_src = FileUtils::joinPaths($rainbow_grades_dir, 'manual_customization.json');
+                break;
+            case 'gui':
+                $customization_src = FileUtils::joinPaths($rainbow_grades_dir, 'gui_customization.json');
+                break;
+            default:
+                $msg = 'Invalid request: Unknown selected value.';
+                return JsonResponse::getErrorResponse($msg);
+        }
+
+        // Copy the source file to the destination
+        if (!copy($customization_src, $customization_dest)) {
+            $msg = 'File copy failed: Could not copy file.';
+            return JsonResponse::getErrorResponse($msg);
+        }
+
+        $msg = 'Rainbow Grades Customization set successfully';
+
+        return JsonResponse::getSuccessResponse(['status' => 'success', 'selected_value' => $selectedValue, 'message' => $msg ]);
```

Comment:
It's generally better to put each element of an array on a separate line so the diff is cleaner if someone comes along later and changes a single element.  It's a good idea to use a trailing comma for the last element of the array for the same reason.
```suggestion
        return JsonResponse::getSuccessResponse([
            'status' => 'success',
            'selected_value' => $selectedValue,
            'message' => $msg,
        ]);
```